### PR TITLE
Align fuchsia and non-fuchsia tracing

### PR DESCRIPTION
**Pull request by dnfield**
Fixes https://github.com/flutter/flutter/issues/33789

This will allow things like flutter driver to work, along with https://dart-review.googlesource.com/c/sdk/+/104948

Sorry for the size of the PR - the vast majority of the changes here are find and replaces for `TRACE_*` to `FML_TRACE_*`.

Key changes are in trace_event.h and dart_vm.cc.

[trace_event.h](https://github.com/flutter/engine/pull/9199/files#diff-4a43a7f813a403416e37ff04e754a66cL13):

- Prefix all our trace macros with `FML_`.
- Remove special casing for Fuchsia for these macros. Let Dart take care of forwarding these events to systrace.  This is tested and working locally (both the timeline and the systrace are getting trace events this way).

[dart_vm.cc](https://github.com/flutter/engine/pull/9199/files#diff-b2f207848a5623d17377178d566274a9R345):

- Don't force fuchsia to do systrace only - only if the VM has been told to do systrace specifically. This means that by default now the engine will foward things over to Dart for tracing, which will take care of forwarding things over to systrace.

The linked bug I'm removing a reference to (DNO-448) is marked as resolved in Jira.

### CODE REVIEW

**Comment by dnfield** [comment]
From offline conversation: it would be nice to rework these macros so they take variadic arguments and we can get rid of TRACE_FOO1/TRACE_FOO2. etc.

**Comment by chinmaygarde** [comment]
I don't think your editor is setup to format on save when doing a search and replace. In case you are using Sublime, Sublime Hooks with the Clang Format plugin works well for me.

File: runtime/dart_vm.cc [hunk 1]
```diff
@@ -351,11 +354,8 @@ DartVM::DartVM(std::shared_ptr<const DartVMData> vm_data,
     PushBackAll(&args, kDartTraceStartupArgs, fml::size(kDartTraceStartupArgs));
   }
 
-#if defined(OS_FUCHSIA)
-  PushBackAll(&args, kDartFuchsiaTraceArgs, fml::size(kDartFuchsiaTraceArgs));
-  PushBackAll(&args, kDartTraceStreamsArgs, fml::size(kDartTraceStreamsArgs));
-#endif
-
+  // PushBackAll(&args, kDartTraceStreamsArgs,
```
Comment (zanderso) [inline runtime/dart_vm.cc]: rm commented out code.

Comment (zanderso) [reply runtime/dart_vm.cc]: replace with a prose comment or TODO if needed.

Comment (dnfield) [reply runtime/dart_vm.cc]: Done

**Comment by dnfield** [comment]
I'm going to wait to land this until the related change in Dart is confirmed as good.

**Pull request merged**